### ISSUE TITLE
[v1.0.0b] - Ingredient sections lost after parsing

### ISSUE THREAD

**Issue by Miroito** [mealie-recipes/mealie#1232]
### First Check

- [X] This is not a feature request
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.

### What is the issue you are experiencing?

Sections created prior to ingredient parsing (mostly on recipes created via url) are lost after the ingredient parse operation.
To reproduce:
1. Create a recipe from a url
2. Edit it and create some random sections
3. Save
4. Edit again and click on parse to go to the ingredient parsing screen
5. here we can already see that the sections created before are gone
6. Modify the ingredients using the info from the parser and click Save All
7. Back in the recipe screen the sections are still gone

### Deployment

Other

### Deployment Details

Does not depend on deployment. I have tested this bug on my deployed instance and my development instance.

**Comment by michael-genson** [comment]
I looked into this a bit and identified the issue. I tinkered with the JS for 20 minutes or so and couldn't come up with a fix, might try to crack it later.

The attribute that's being lost is the title on the RecipeIngredient, which is where the section is stored:
```python
class RecipeIngredient(SqlAlchemyBase, BaseMixins):
    __tablename__ = "recipes_ingredients"
    id = Column(Integer, primary_key=True)
    position = Column(Integer)
    recipe_id = Column(GUID, ForeignKey("recipes.id"))

    title = Column(String)  # <-- here
    note = Column(String) 
```

The implementation first extracts the recipe string by pulling it from the note in the frontend, sends that string to the backend, and the backend spits out a new ingredient object. Since we're only running the note through the API, not the whole ingredient, the title is lost:
```javascript
// =========================================================
// Parser Logic
const parser = ref<Parser>("nlp");
const parsedIng = ref<ParsedIngredient[]>([]);

async function fetchParsed() {
  if (!recipe.value || !recipe.value.recipeIngredient) {
    return;
  }
  const raw = recipe.value.recipeIngredient.map((ing) => ing.note ?? "");  // <-- here, "raw" is a string array of ingredient notes
  const { data } = await api.recipes.parseIngredients(parser.value, raw);  // an array of new ingredient objects without the titles
```

It's _probably_ an easy fix to loop through the unparsed ingredient array and map the title attribute over to the new parsed array, but I'm not great with JS.

**Comment by michael-genson** [comment]
This should be fixed in PR #1368
